tests/codex/sales/testsales.nim

```diff
@@ -188,9 +189,14 @@ asyncchecksuite "Sales":
     await repoTmp.destroyDb()
     await metaTmp.destroyDb()
 
+  proc isInState(idx: int, state: string): Future[bool] {.async.} =
```

Comment:
Does this need to be async?

Reply:
Yes: check-eventually is needed to give sales time to load the agent.

Reply:
Right, but shouldn't we just early return if `sales.agent.len == 0`? I don't think there's a need for `check eventually` in the `isInState` proc, since its usage is used in a `check eventually isInState...`.

Reply:
Creating a PR for this: https://github.com/codex-storage/nim-codex/pull/1021